cmd/signature-v4-postpolicyform.go

```diff
@@ -152,19 +149,17 @@ func parsePostPolicyFormV4(policy string) (PostPolicyForm, error) {
 }
 
 // checkPostPolicy - apply policy conditions and validate input values.
-func checkPostPolicy(formValues map[string]string) APIErrorCode {
-	/// Decoding policy
-	policyBytes, err := base64.StdEncoding.DecodeString(formValues["Policy"])
-	if err != nil {
-		return ErrMalformedPOSTRequest
-	}
-	postPolicyForm, err := parsePostPolicyFormV4(string(policyBytes))
-	if err != nil {
-		return ErrMalformedPOSTRequest
-	}
+func checkPostPolicy(formValues map[string]string, postPolicyForm PostPolicyForm) APIErrorCode {
 	if !postPolicyForm.Expiration.After(time.Now().UTC()) {
 		return ErrPolicyAlreadyExpired
 	}
+	if postPolicyForm.Conditions.ContentLengthRange.Max > maxObjectSize {
+		// Upload object size can not be more than what is allowed.
+		return ErrNotImplemented
```

Comment:
aws-s3 is not doing this check, returns error only if the size of the actual uploaded object is > 5G, we can do it the same way.